paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java

```diff
@@ -826,7 +826,12 @@ public final CatalogPartition getPartition(
     @Override
     public final boolean partitionExists(ObjectPath tablePath, CatalogPartitionSpec partitionSpec)
             throws CatalogException {
-        return false;
+        try {
+            List<CatalogPartitionSpec> partitionSpecs = getPartitionSpecs(tablePath, partitionSpec);
```

Comment:
Here we need an effective method to check partition. For example, we can provide a `SnapshotReaderImpl.withPartitionFilter` to optimize scanning.

Reply:
I updated it